### PR TITLE
Remasc federation config bug

### PR DESCRIPTION
**Pull request by adauto82**
There was an issue with the construction of briedge support that didn't set the correct constants.

### CODE REVIEW

**Comment by rskops** [comment]
SonarQube analysis reported 150 issues
* ![MAJOR][MAJOR] 92 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 23 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L373](https://github.com/rsksmart/rskj/blob/ec6a357ce0cb94741b58a7deb9ef55cf810a29d6/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L373): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/remasc/Remasc.java [hunk 1]
```diff
@@ -137,7 +138,15 @@ void processMinersFees() throws IOException, BlockStoreException {
         // the update of bytes is needed, because BridgeSupport creation could alter
         // the storage when getChainHead is null (specially in production)
         processingRepository = processingRepository.startTracking();
-        RemascFederationProvider federationProvider = new RemascFederationProvider(processingRepository);
+        BridgeSupport bridgeSupport = new BridgeSupport(
+                repository,
```
Comment (diega) [inline rskj-core/src/main/java/co/rsk/remasc/Remasc.java]: I think this should be the `processingRepository` as @ajlopezrsk suggested offline

**Comment by rskops** [comment]
SonarQube analysis reported 150 issues
* ![MAJOR][MAJOR] 92 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 23 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L373](https://github.com/rsksmart/rskj/blob/e7509648f5201e0c092a1dd5fa9c148b034ca451/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L373): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by rskops** [comment]
SonarQube analysis reported 148 issues
* ![MAJOR][MAJOR] 91 major
* ![MINOR][MINOR] 35 minor
* ![INFO][INFO] 22 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [Bridge.java#L1](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L1): Class co.rsk.peg.Bridge defines static field "co.rsk.peg.Bridge.functionCostMap" which appears to allow memory bloat [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3APMB_POSSIBLE_MEMORY_BLOAT)
1. ![MAJOR][MAJOR] [Bridge.java#L267](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L267): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [Bridge.java#L298](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L298): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L308](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L308): Extract this nested try block into a separate method. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1141)
1. ![MAJOR][MAJOR] [Bridge.java#L313](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L313): Return an empty array instead of null. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1168)
1. ![MAJOR][MAJOR] [Bridge.java#L330](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L330): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L334](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L334): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L342](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L342): Remove this unused method parameter "args". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [Bridge.java#L350](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L350): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)
1. ![MAJOR][MAJOR] [Bridge.java#L373](https://github.com/rsksmart/rskj/blob/c9b1346f6b2c9e7c11c76c5f28dde4648494bf46/rskj-core/src/main/java/co/rsk/peg/Bridge.java#L373): Define and throw a dedicated exception instead of using a generic one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00112)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by tinchou** [comment]
@diega changes were made. Is this OK now?

File: rskj-core/src/main/java/co/rsk/remasc/RemascFederationProvider.java [hunk 2]
```diff
@@ -18,8 +12,8 @@
 public class RemascFederationProvider {
     private BridgeSupport bridgeSupport;
 
-    public RemascFederationProvider(Repository repository) throws IOException, BlockStoreException {
-        this.bridgeSupport = new BridgeSupport(repository, PrecompiledContracts.BRIDGE_ADDR, null, null, null, null);
+    public RemascFederationProvider(BridgeSupport bridgeSupport) throws IOException, BlockStoreException {
```
[the remark this answers is not in the document]
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/remasc/RemascFederationProvider.java]: I think someone could argue that the abstraction you mention should be in `BridgeSupport` and not in `RemascFederationProvider`.

In the end, both approaches are valid, but I'd prefer to keep instantiations to the bare minimum. Whenever a class has a `new XXX` you are limiting testing capabilities and tying your class to the `XXX` class implementation (and not to its interface). Also notice that we're passing lots of `null`s to `BridgeSupport`: that is a sign that we're doing something wrong.

If we can make `RemascFederationProvider` not instantiate anything I think that's a win, and we can fix `Remasc` later on.

**Pull request merged**